test/services/auth.test.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @jest-environment node
+ */
+
+import { faker } from '@faker-js/faker';
+
+import { authenticateLogin } from '../../src/services/auth';
+import { user } from '../factories/user';
+import { createUser } from '../../src/app/actions/user';
+
+describe('Auth:', () => {
+  describe('given correct credentials', () => {
+    it('returns user', async () => {
+      const userPayload = { name: user.name, email: user.email, password: user.password };
+      await createUser(userPayload);
+      const data = await authenticateLogin({
+        email: user.email,
+        password: user.password,
+      });
+
+      expect(data).not.toBeNull();
+      expect(data.email).toEqual(user.email);
+      expect(data.name).toEqual(user.name);
+    });
+  });
+
+  describe('when there is no user as credential', () => {
+    it('throws bad request error.', async () => {
+      await expect(
+        authenticateLogin({ email: 'dev@nimblehq.co', password: 'random' })
+      ).rejects.toHaveProperty('code', 400);
+    });
+  });
+
+  describe('when no credentials are passed', () => {
+    it('throws bad request error', async () => {
+      await expect(authenticateLogin({ email: '', password: '' })).rejects.toHaveProperty(
+        'code',
+        400
+      );
+    });
+  });
+
+  describe('when passed password is incorrect', () => {
+    it('throws bad request error', async () => {
+      const email = faker.internet.email();
+      const userPayload = {
+        name: user.name,
+        email,
+        password: user.password,
+      };
+      await createUser(userPayload);
+
+      await expect(
+        authenticateLogin({
+          email: email,
```

Comment:
I recommend to set it to a fixed email address:

```suggestion
          email: 'other.user@nimblehq.co',
```

Reply:
Fixed in https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/66/commits/60c2f10bae5e1863561ada4099a6b2bacb68772b